phoenix-core/src/it/java/org/apache/phoenix/end2end/BackwardCompatibilityIT.java

```diff
@@ -118,6 +125,7 @@ public synchronized void doSetup() throws Exception {
         zkQuorum = "localhost:" + hbaseTestUtil.getZkCluster().getClientPort();
         url = PhoenixRuntime.JDBC_PROTOCOL + PhoenixRuntime.JDBC_PROTOCOL_SEPARATOR + zkQuorum;
         DriverManager.registerDriver(PhoenixDriver.INSTANCE);
+        checkForPreConditions();
```

Comment:
We need to have this before each test as it checks for the non existence of any SYSTEM tables before each test run to make sure that there is no existing metadata

Reply:
Yes that's what is happening. This is now called in `doSetup()` which will be invoked before each test i.e. it is marked as `@Before`

Reply:
Makes sense, thanks